Task: Add jsx compiler option to editor-host-client tsconfig

## Description

Add `"jsx": "react"` to `packages/editor-host-client/tsconfig.json` compiler options to fix TypeScript error TS6142 when resolving `.tsx` files from `editor-renderer-react-flow`.

## Acceptance Criteria

- `packages/editor-host-client/tsconfig.json` has `"jsx": "react"` in `compilerOptions`
- The jsx mode matches `editor-renderer-react-flow` (uses `React.createElement`)

## Relevant Files

- `packages/editor-host-client/tsconfig.json` — add `"jsx": "react"` to `compilerOptions`

## Context

`editor-host-client` imports from `@sw-editor/editor-renderer-react-flow`, which exports `.tsx` source files directly. TypeScript resolves these with `editor-host-client`'s tsconfig, which lacks a `jsx` setting. All 6 errors are:
```
error TS6142: Module './nodes/EndNode.js' was resolved to '.../EndNode.tsx', but '--jsx' is not set.
```

<!-- agent-meta
entity: task
source_feature: 195
source_plan: 196
depends_on: []
-->

--- a/packages/editor-host-client/tsconfig.json
+++ b/packages/editor-host-client/tsconfig.json
@@ -3,7 +3,8 @@
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
     "rootDir": "src",
-    "outDir": "dist"
+    "outDir": "dist",
+    "jsx": "react"
   },
   "include": ["src"]
 }

